### PR TITLE
Fix colors / styling for new query results pane

### LINKED ISSUE #18249: Improve styling of data export buttons in query results sidebar
Image

We're using `appearance: subtle` other places to make it look better:

https://react.fluentui.dev/?path=/docs/components-button-button--docs#appearance

Image
Image

### LINKED ISSUE #18293: Context menu in results grid doesn't appear correctly themed on Light mode
OS: macos
Steps to Reproduce:

1. RIght click on result grid in light mode,

Expected: Correct theme of menu

Actual: Dark themed menu in Light mode

Image

### PR DESCRIPTION
**Pull request by cssuh**
Fixed context menu styling:
Screenshot 2024-10-24 090108

Fixed sort menu styling:
Screenshot 2024-10-24 090115

Changed button appearance:
Screenshot 2024-10-24 090144

Tested in both dark mode & light mode.

Also fix context menu behavior - 

contextMenu

Before: 
Clicking dropdown arrow again wouldn't close context menu

After:
Clicking dropdown arrow again closes context menu, clicking other dropdown arrows closes current context menu and opens a new one

Fixes #18293
Fixes #18249

### CODE REVIEW

**Comment by caohai** [comment]
The `Cancel` option in the menu list is a bit confusing but since you are currently working on the filtering menu, we can revisit this later.
ADS:
image

**Comment by caohai** [comment]
image

1. The sorting menu is a bit too wide for the text
2. Is it intended to allow both sorting menu and context menu to show up at the same time? They might get in the way of each other. Do we want to close one if the other is opened?

**Comment by cssuh** [comment]
> The `Cancel` option in the menu list is a bit confusing but since you are currently working on the filtering menu, we can revisit this later. ADS: image

Yes I'm planning to clean up this whole menu when I add in the actual filters here.

File: src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts [hunk 1]
```diff
@@ -117,14 +117,24 @@ export class HeaderFilter<T extends Slick.SlickData> {
...
+            if (isSameButton) {
+                // If clicking the same button, close the popup and reset activePopup
+                this.activePopup.fadeOut();
+                this.activePopup = null;
+                return; // Exit since we're just closing the popup
+            } else {
+                // If it's a different button, close the current popup before opening the new one
+                this.activePopup.fadeOut();
+                this.activePopup = null;
+            }
```
Comment (Benjin) [inline src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts:133]: May as well combine these and only return if it's the same button

**Review by Benjin — APPROVED** [review]

**Comment by Benjin** [comment]
Mentioning what feature you're fixing the colors/styles for in the PR title would be good for when people read through commit history in the future.

> Fix colors/styling for new query results pane

**Review by lewis-sanchez — APPROVED** [review]

**Comment by cssuh** [comment]
> image
> 1. The sorting menu is a bit too wide for the text
> 2. Is it intended to allow both sorting menu and context menu to show up at the same time? They might get in the way of each other. Do we want to close one if the other is opened?

Created #18336 to track this, will address in a separate PR

**Pull request merged**